lesson3_videos.py

```diff
@@ -35,27 +35,47 @@ def show(self):
         print(self.person_name, self.lives_in_town, self.nickname)
 
 
+class Department(BaseModel):
+    """
+        This class defines Department, which maintains details of in
+        which Department a Person held a Job
+    """
+
+    dept_number = pw.CharField(primary_key=True, max_length=30)
+    dept_name = pw.CharField(max_length=30)
+    dept_manager = pw.ForeignKeyField(Person, related_name='person', null=False)
+
+    def show(self):
+        """ display an instance """
+        print(self.dept_number, self.dept_name, self.dept_manager)
+
+
 class Job(BaseModel):
     """
         This class defines Job, which maintains details of past Jobs
         held by a Person.
     """
 
-    job_name = pw.CharField(primary_key = True, max_length = 30)
-    start_date = pw.DateField(formats = 'YYYY-MM-DD')
-    end_date = pw.DateField(formats = 'YYYY-MM-DD')
+    job_name = pw.CharField(primary_key=True, max_length=30)
+    start_date = pw.DateField(formats='YYYY-MM-DD')
+    end_date = pw.DateField(formats='YYYY-MM-DD')
 
-    salary = pw.DecimalField(max_digits = 7, decimal_places = 2)
+    salary = pw.DecimalField(max_digits=7, decimal_places=2)
     person_employed = pw.ForeignKeyField(
-        Person, related_name='was_filled_by', null = False)
+        Person, related_name='was_filled_by', null=False)
+    dept_name = pw.ForeignKeyField(Department, related_name='dept_name', null=False)
```

Comment:
Change _related_name='dept_name'_ to _related_name='jobs'_, that's what's forcing a NULL into _dept_name_ when creating a new department. _related_name_ or the newer version, _backref_ are a bit confusing in my opinion. Here's a good discussion on the topic:

https://github.com/coleifer/peewee/issues/2027